File: web-client/src/app/Models/Dorm.ts
import { IReview } from './Review';
import { ReviewSummary } from './ReviewSummary';

interface IBaseDorm {
  id: number;
  name: string;
  style: string;
  cost: string;
  reviews: IReview[];
  reviewSummary: ReviewSummary | null;
}

export interface ILargeDorm extends IBaseDorm {
  description: string;
  nameQueryParameter: string;
}

export interface ISmallDorm extends IBaseDorm {
  nameQueryParameter: string;
  image: string;
}

export type Dorm = IBaseDorm;

File: api/Models/Dorm.cs
namespace api.Models;

using System.ComponentModel.DataAnnotations;

public class Dorm : IReviewable
{
    [Key]
    public int Id { get; set; }

    public string Name { get; set; }
    public string Style { get; set; }
    public string Cost { get; set; }
    public string Description { get; set; }
    public string NameQueryParameter { get; set; }

    public virtual List<Review> Reviews { get; set; } = [];
    public virtual ReviewSummary ReviewSummary { get; set; }

}